### PR TITLE
Remove lodash

### PR DESCRIPTION
**Pull request by marcosbc**
Lodash is not an essential dependency, but it is used extensively in spite of there being native functions achieving the same things. We want to standarize on native functions (ES6) in order to remove an additional dependency.

Signed-off-by: Marcos Bjoerkelund <marcos@bitnami.com>

### CODE REVIEW

File: lib/core/build-manager/build-environment/compilation-env-handler.js [hunk 1]
```diff
@@ -11,7 +9,7 @@ const _ = require('nami-utils/lodash-extra');
  */
 class CompilationEnvVarsHandler {
   constructor(options) {
-    options = _.opts(options, {platform: null});
+    options = Object.assign({platform: null}, options);
```
Comment (javsalgar) [inline lib/core/build-manager/build-environment/compilation-env-handler.js:12]: Is platform: null necessary?

File: lib/core/build-manager/build-environment/compilation-env-handler.js [hunk 2]
```diff
@@ -129,24 +127,24 @@ class CompilationEnvVarsHandler {
...
+    options = Object.assign({stringify: true}, options);
     let values = null;
-    if (_.isEmpty(extraVars)) {
+    if (Object.entries(extraVars || {}).length === 0) {
       values = this._environmentVars;
     } else {
       values = this._environmentVars;
-      _.each(extraVars, (value, name) => {
-        if (_.has(values, name)) {
+      extraVars.forEach(([name, value]) => {
```
Comment (javsalgar) [inline lib/core/build-manager/build-environment/compilation-env-handler.js:136]: This is an object, right?

File: lib/core/build-manager/component-provider/index.js [hunk 3]
```diff
@@ -18,9 +17,9 @@ const RecipeLogicProvider = require('./recipe/logic-provider');
  */
 class ComponentProvider {
   constructor(componentTypeCollections, options) {
-    options = _.opts(options, {
+    options = Object.assign({
       logger: null,
```
Comment (javsalgar) [inline lib/core/build-manager/component-provider/index.js:21]: Is  logger: null necessary?

**Review by javsalgar — COMMENTED** [review]
Added some comments

File: lib/core/build-manager/build-environment/compilation-env-handler.js (hunk 2 above)
Comment (marcosbc) [reply lib/core/build-manager/build-environment/compilation-env-handler.js]: Yes it is, this is a typo, fixed now.

File: lib/core/build-manager/component-provider/index.js (hunk 3 above)
Comment (marcosbc) [reply lib/core/build-manager/component-provider/index.js:21]: Not necessary, but I think it is good to initialize all supported options by this method.

File: lib/core/build-manager/build-environment/compilation-env-handler.js (hunk 1 above)
Comment (marcosbc) [reply lib/core/build-manager/build-environment/compilation-env-handler.js:12]: Not necessary, but I think it is good to initialize all supported options by this method.

**Review by javsalgar — APPROVED** [review]

**Pull request merged**